dora/tests/integration/src/test/java/alluxio/client/fs/io/FileInStreamIntegrationTest.java

```diff
@@ -463,220 +411,9 @@ public void positionedReadWithoutCaching() throws Exception {
-            boolean achieved = true;
-            // Expect only second block to be cached, other blocks should be empty in Alluxio
-            for (int i = 0; i < st.getFileBlockInfos().size(); i++) {
-              FileBlockInfo info = st.getFileBlockInfos().get(i);
-              if (i == 1) {
-                achieved = achieved && !info.getBlockInfo().getLocations().isEmpty();
-              } else {
-                achieved = achieved && info.getBlockInfo().getLocations().isEmpty();
-              }
-            }
-            return achieved;
-          } catch (Exception e) {
-            return false;
-          }
-        });
-      } else {
-        Thread.sleep(1000);
-        status = mFileSystem.getStatus(uri);
-        Assert.assertEquals(0, status.getInAlluxioPercentage());
-      }
-    }
-  }
-
-  @Test(timeout = 10000)
-  public void asyncCacheFirstBlockPRead() throws Exception {
-    String filename = mTestPath + "/file_" + MAX_LEN + "_" + mWriteUnderStore.hashCode();
-    AlluxioURI uri = new AlluxioURI(filename);
-
-    for (ReadType readType : ReadType.values()) {
-      mFileSystem.free(uri);
-      CommonUtils.waitFor("No in-Alluxio data left from previous iteration.", () -> {
-        try {
-          URIStatus st = mFileSystem.getStatus(uri);
-          return st.getInAlluxioPercentage() == 0;
-        } catch (Exception e) {
-          return false;
-        }
-      });
-      FileInStream is = mFileSystem.openFile(uri,
-          OpenFilePOptions.newBuilder().setReadType(readType.toProto()).build());
-      // Positioned reads trigger async caching after reading and do not need to wait for a close
-      // or a block boundary to be crossed.
-      URIStatus status = mFileSystem.getStatus(uri);
-      Assert.assertEquals(0, status.getInAlluxioPercentage());
-      is.positionedRead(BLOCK_SIZE / 2, new byte[1], 0, 1);
-      if (readType.isCache()) {
-        CommonUtils.waitFor("First block to be cached.", () -> {
-          try {
-            URIStatus st = mFileSystem.getStatus(uri);
-            boolean achieved = true;
-            // Expect only first block to be cached, other blocks should be empty in Alluxio
-            for (int i = 0; i < st.getFileBlockInfos().size(); i++) {
-              FileBlockInfo info = st.getFileBlockInfos().get(i);
-              if (i == 0) {
-                achieved = achieved && !info.getBlockInfo().getLocations().isEmpty();
-              } else {
-                achieved = achieved && info.getBlockInfo().getLocations().isEmpty();
-              }
-            }
-            return achieved;
-          } catch (Exception e) {
-            return false;
-          }
-        });
-      } else {
-        Thread.sleep(1000);
-        status = mFileSystem.getStatus(uri);
-        Assert.assertEquals(0, status.getInAlluxioPercentage());
-      }
-      is.close();
-    }
-  }
-
-  @Test
-  public void syncCacheFirstBlock() throws Exception {
```

Comment:
@dbw9580 do you know if this test should still be relevant? I figure it's the `getFileBlockInfos()` is somehow a deprecated API? But we should still fix this one?

Reply:
it basically requires the first portion of the file is cached when the length of the read is larger than the size of a block. The requirement still applies to page store, but it may not be able to report block level information. I suggest to use the cache usage API on worker's cache manager directly to verify the size of the file being cached. something like

```java
CacheManager cm = worker.getCacheManager(); // <- this method may not exist yet
long usedBytes = cm.getUsage()
    .partitionedBy(file(uri))
    .map(CacheUsage::used)
    .orElse(0);
assertEquals(data.length, usedBytes)
```

Reply:
let's remove this test for now as it's still using Block API. We can add a new one.